enhancements/gep/001-loadbalancer/README.md

```diff
@@ -0,0 +1,119 @@
+# Gravity Enhancement Proposal
+
+## GEP-001: Change DNS balancer to a software loadbalancer (HAProxy)
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+    - [User Stories (Optional)](#user-stories-optional)
+        - [Story 1](#story-1)
+        - [Story 2](#story-2)
+    - [Notes/Constraints/Caveats (Optional)](#notesconstraintscaveats-optional)
+    - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+    - [Test Plan](#test-plan)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+    - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+- [Implementation History](#implementation-history)
+
+<!-- /toc -->
+
+## Summary
+The main goal of this proposal is to replace dns balancing with a software loadbalancer.
+Thus, all requests to the kube-apiserver from Kubernetes components (kubelet, scheduler, control-manager, kube-proxy)
+will go through this loadbalancer. Also, requests to the docker registry must also go through the loadbalancer.
+
+## Motivation
+At the moment, Gravity is using CoreDNS to determine which master to send requests to,
+but DNS is not a good tool for balancing traffic.
+This proposal aims to provide the following improvements:
+  1. Evenly distribute the load between all apiserver replicas
+  1. Improve the stability and scalability of the cluster
+  1. Nodes will react faster to master failures
+  1. Cluster upgrades will be faster and more efficient
+
+### Goals
+When creating a cluster, the user can choose the load balancer to be either external or internal.
```

Comment:
It's not clear to me what the desire to support this with external load balancers is, but that could be internal requirements. However, I would just caution being careful with supporting lots of different models, if the built-in load balancer will sufficiently support the needs.